sympy/core/power.py

```diff
@@ -383,12 +383,12 @@ def inverse(self, argindex=1):
         return None
 
     @property
-    def base(self):
-        return self._args[0]
+    def base(self) -> Expr:
+        return self.args[0]
```

Comment:
Maybe it would be better to declare that the args of `Pow` are `Expr`.

Reply:
Yes, it is already there
https://github.com/sympy/sympy/blob/641a8b2839f0756808fb4cf941b82955f527a231/sympy/core/power.py#L278

Reply:
So why does pyright think that it's Basic?

Reply:
If I only add return type hint and keep function body as `return _args[0]` then pyright will complain, as it doesn’t know `_args` and `args` is the same thing.
Now it doesn’t complain.

Reply:
Maybe there should also be a type-hint for `Pow._args` then.

Reply:
_args isn’t used elsewhere in this file, and its usage (for both developer and user) is not encouraged either. It’s more like an internal representation of args for Basic and not its subclasses.

Reply:
What if instead we define the type in `def __new__(cls, b, e, evaluate):`? Shouldn't it be able to figure that the values in `_arg` are actually `Expr` instead of `Basic`?

Reply:
> What if instead we define the type in `def __new__(cls, b, e, evaluate):`? Shouldn't it be able to figure that the values in `_arg` are actually `Expr` instead of `Basic`?

This `__new__` function doesn't always return a `Pow` object, like `Pow(1,1)` returns `Integer(1)`. And it delegates to `Expr.__new__` for creating `Pow` object (and after that we still don't know if it's a `Pow`).
https://github.com/sympy/sympy/blob/641a8b2839f0756808fb4cf941b82955f527a231/sympy/core/power.py#L372-L377
Given the complexity, I don't think pyright can track all the way down to know `_arg` is `Expr`.

Reply:
It looks to me like `_args` is being used rather than `args` because it is maybe slightly faster. I'm not sure we should change that.

Reply:
The usage of `_args` here dates back to the first git commit of sympy in 2007. Later in 2008, @certik made the note in https://github.com/sympy/sympy/commit/b89a24efab8e4e25aed2ecb567905a1720b4bb62 that `args` should always be used. Could @certik or @asmeurer please give some advices?

Reply:
I just don't think it's necessary to change this. As far as I can tell all that is needed is a type-hint on `_args` to clarify that `Pow._args` is `tuple[Expr, Expr]`.